src/Illuminate/Foundation/Console/KeyGenerateCommand.php

```diff
@@ -111,6 +119,40 @@ protected function writeNewEnvironmentFileWith($key)
         return true;
     }
 
+    /**
+     * Move the current APP_KEY to APP_PREVIOUS_KEYS.
+     *
+     * @param  string  $envContent
+     * @param  string  $currentKey
+     * @return string
+     */
+    protected function moveToPreviousKeys($envContent, $currentKey)
+    {
+        preg_match('/^APP_PREVIOUS_KEYS=(.*)$/m', $envContent, $matches);
+
+        $existingPreviousKeys = isset($matches[1]) ? trim($matches[1]) : '';
+
+        $previousKeysArray = array_filter(explode(',', $existingPreviousKeys));
+        $previousKeysArray[] = $currentKey;
+        $newPreviousKeys = implode(',', $previousKeysArray);
+
+        if (preg_match('/^APP_PREVIOUS_KEYS=/m', $envContent)) {
+            $envContent = preg_replace(
+                '/^APP_PREVIOUS_KEYS=.*$/m',
```

Comment:
Technically, you don't need yet another RegExp here

```suggestion
        if (preg_match('/^APP_PREVIOUS_KEYS=/m', $envContent)) {
            $envContent = str_replace(
                $matches[0],
```
or

```suggestion
        if (preg_match('/^APP_PREVIOUS_KEYS=/m', $envContent, $matches)) {
            $envContent = str_replace(
                $matches[0],
```
will probably be enough